batcher/aligned-batcher/src/lib.rs

```diff
@@ -675,28 +728,31 @@ impl Batcher {
         // pad leaves to next power of 2
         let padded_leaves = Self::pad_leaves(leaves);
 
-        let call = self.payment_service.create_new_task(
+        match try_create_new_task(
             batch_merkle_root,
-            batch_data_pointer,
-            padded_leaves,
-            signatures,
+            batch_data_pointer.clone(),
+            padded_leaves.clone(),
+            signatures.clone(),
             gas_for_aggregator,
             gas_per_proof,
-        );
-
-        info!("Creating task for: {}", hex::encode(batch_merkle_root));
-
-        let pending_tx = call
-            .send()
-            .await
-            .map_err(|e| BatcherError::TaskCreationError(e.to_string()))?;
-
-        let receipt = pending_tx
-            .await
-            .map_err(|_| BatcherError::TransactionSendError)?
-            .ok_or(BatcherError::ReceiptNotFoundError)?;
-
-        Ok(receipt)
+            &self.payment_service,
+        )
+        .await
+        {
+            Ok(receipt) => Ok(receipt),
+            Err(_) => {
```

Comment:
I think we should check the reason of the error. For example, if the transaction was reverted it wouldn't make sense to retry with the fallback and we will be paying two times. We should try with the fallback only when the error is related to RPC connection/reliability issues